doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,261 @@
+scorecard v1alpha2 Test Result output.  This allows scorecard to parse
+the custom test log output and aggregate the results like any other
+scorecard test.
+
+The scorecard expected output would be in JSON, and would match the following definition:  https://github.com/operator-framework/operator-sdk/blob/master/pkg/apis/scorecard/v1alpha2/types.go#L36
+
+The scorecard v1alpha2 Test Result output contains information such as
+pass or fail status and suggestions/errors produced by the test that
+the end user would want to see in the aggregated scorecard results.
+
+
+### Custom Test Development Model
+
+The developer of a custom test would have a workflow similar to:
+ 
+ * create a Pod manifest to run their custom test image, copying a sample Pod manifest from the scorecard examples in the SDK repository
+ * write a custom test binary in the language of their choice, sample custom test within the SDK repository would be available as a starting point
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * when the test binary works as expected, the developer would build and push the test image to a container registry accessible to the scorecard test environment
+ * the scorecard configuration would be updated to reference the custom test Pod manifest location
+ * the scorecard would discover the custom test based on the scorecard configuration and also the custom test labels applied by the end-user within the custom test Pod manifest
+
+### Custom Test Packaging and Configuration
+
+In this design proposal, custom tests are Pod manifests in YAML that run a custom test image.
+
+Note that it would also be possible to add a resource limit to the scorecard configuration as shown in the example below.  Those resource limits would be applied to custom tests that are executed allowing for some control as to how much resources are available for the test execution.
+
+Custom tests (i.e. Pod manifests) would be accessible from a local disk location.  Custom test locations would be configured from within the scorecard configuration 
+file.  Custom test labels would be also in the scorecard configuration file whereas the scorecard could pick tests to run by test labels.  For example, the scorecard configuration might be as follows to indicate the presence of custom tests from different sources:
+```
+scorecard:
+  # Setting a global scorecard option
+  output: text
+  tests:
+    init-timeout: 60
+    resources:
+      limits:
+        cpu: "1"
+        memory: "200Mi"
+      requests:
+        cpu: 750m
+        memory: "100Mi"
+      test-path:
+        - "tests/here/ondisk"
+        - "someisv/tests"
+        - "operator-sdk/scorecard-tests"
+      cr-manifest:
+        - "deploy/crds/cache.example.com_v1alpha1_memcached_cr.yaml"
+        - "deploy/crds/cache.example.com_v1alpha1_memcached_cr2.yaml"
```

Comment:
Is it reasonable to remove the `cr-manifest` configuration and require the CR manifests under test to be included in the CSV examples?

Reply:
[This proposal](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/contributors/design-proposals/internal-objects.md#hiding-internal-concepts-from-end-users) is also relevant. If a CRD is listed as data-only, we should not make the same assumptions that we would about workload CRDs.

I'm not sure how prevalent usage of these annotations are yet, so maybe something that can be considered in a follow-on.

Reply:
Seems reasonable to have the scorecard drop cr-manifest from its scorecard config and instead just pull them from the CSV examples.  Probably this works is done with the 'make bundle mandatory' story.  Same applies for the scorecard config's csv-manifest, that would no longer be used, but instead the CSV within the bundle.

The annotation proposal is interesting, I think its something we would need to address when that feature is available based on my reading of that.